### DISCUSSION TITLE
Background workers

### DISCUSSION THREAD

**Discussion by Minivera** [encoredev/encore#70] [Help] [answered]
I have some background tasks I want to run on a regular basis to sync data when users authenticate with my application. For testing purposes, its currently implemented as a goroutine that's started in the auth endpoint, but that's not working great. I tried to check the source code to see if there was a way to start some routine when a service starts so it could do things like poll a database for tasks to run, but I couldn't find anything.

Is there any way at the moment to hook into a service's main function to start a go routine or to run an async loop of some kind?

**Comment by eandre** [comment, marked as answer]
Hey @Minivera, great question! We’re going to add much better support for background tasks, cron jobs, and so on. Right now the best solution tends to be to implement a cron job outside of Encore that pings an endpoint on a regular basis. Is that something that would work for your use case? It would be very useful to understand more about what you’re trying to do :)

**Reply by Minivera** [reply]
That would work yes! I was wondering if encore had some hidden functionality I could use instead since cronjobs can be a pain in local dev. My use case is to implement [device OAuth with Github](https://docs.github.com/en/developers/apps/building-oauth-apps/authorizing-oauth-apps#device-flow). Users would call the SignIn API endpoint and get a device code to enter on GitHub as the response. Since the endpoint resolves when it returns, I wanted to start a task that calls the GitHub token API for an access token using the device code on a regular interval (every 5 seconds for, at most, 15 minutes).

Since this polling could potentially put a user account into an invalid state, I though I'd have the tasks saved in the database so they're not lost if the server goes down. Foe example, I could have a loop that polls a db table every second and processes tasks that are ready to be processed with retry implemented since I need to poll GitHub every 5 seconds for an access token.
